src/Stratis.Bitcoin.Features.Consensus/Rules/ProvenHeaderRules/ProvenHeaderCoinstakeRule.cs

```diff
@@ -48,19 +47,9 @@ public override void Initialize()
             this.stakeValidator = this.PosParent.StakeValidator;
         }
 
-        /// <inheritdoc/>
-        public override void Run(RuleContext context)
+        /// <inheritdoc />
+        protected override void ProcessRule(PosRuleContext context, ChainedHeader chainedHeader, ProvenBlockHeader header)
         {
-            Guard.NotNull(context.ValidationContext.ChainedHeaderToValidate, nameof(context.ValidationContext.ChainedHeaderToValidate));
-
-            ChainedHeader chainedHeader = context.ValidationContext.ChainedHeaderToValidate;
-            int height = chainedHeader.Height;
-
-            if (context.SkipValidation || !this.IsProvenHeaderActivated(height))
```

Comment:
Why delete this code?

Reply:
I moved that code, that was common to both 2 PH rules (the other was size rule) and moved to the ProvenHeaderRuleBase.cs
If you look this method is no more the `Run` but another that gets called by ProvenHeaderRuleBase.Run method